node/src/Unison/Runtime/ResourcePool.hs

```diff
@@ -36,8 +35,8 @@ addResourceToMap release pool p r wait maxPoolSize getThread poolSizeR queue = d
       >> STM.atomically (TQ.writeTQueue queue key)
     else release r
 
-_acquire :: (Ord p, H.Hashable p) => (p -> IO r) -> (r -> IO ()) -> Cache p r -> MaxPoolSize -> IORef.IORef Int -> ReaperQueue p -> IO CC.ThreadId -> p -> Int -> IO (r, IO ())
-_acquire acquire release pool maxPoolSize poolSizeR queue getThread p wait = do
+_acquire :: (Ord p, H.Hashable p) => (p -> IO r) -> (r -> IO ()) -> Cache p r -> Int -> MaxPoolSize -> IORef.IORef Int -> ReaperQueue p -> IO CC.ThreadId -> p -> IO (r, IO ())
+_acquire acquire release pool wait maxPoolSize poolSizeR queue getThread p = do
   threadId <- getThread
   m <- M.lookup (p,threadId) pool
```

Comment:
Hey @steveshogren sorry for long delay in getting to this. I'm focused on Unison for a while so should be much more responsive going forward. I'm going to merge as is then stare at the code a bit to figure out a good solution which we can do in a separate PR.